doc/spec_TransferElement.dox

```diff
@@ -385,6 +386,12 @@ This document describes the behaviour of the TransferElement base class, the NDR
     - 5.2.7 A write operation is executed on a transfer element that cannot be written
       - 5.2.7.1 Check with TransferElement::isWriteable() or TransferElement::isReadOnly()
       - \anchor transferElement_C_5_2_7_2 5.2.7.2 Thrown in doPreWrite() [\ref UnifiedTest_TransferElement_C_5_2_7_2 "U"]
+    - 5.2.8 Scalar, 1D or 2D registers of UserType Void
+      - The instantiation of the Void UserType is limited to the respective VoidRegister
+      - Other register types will throw an error upon instantiaion.
+    - 5.2.9 UserType Void in combination with read only and not having wait_for_new_data
```

Comment:
See commments to 5.2.8

- read() is called on a VoidRegisterAccessor without wait_for_new_data()
   - Throws in VoidRegisterAccessor::read() *footnote* (Throwing as early as possible means: before even calling the TransferElement implementation)

Note: It is not thrown in the constructor. If you want a VoidRegisterAccessor for writing you don't need the "wait_for_new_data" flag.

Are we missing a point that if you create a read-only accessor. Ah I completely got that wrong. Use "read-only" with dash. I was reading "read is only useful ..". Again, write the sentence "when an exception is thrown".

We need two points
- Throws is the constructor if read-only VoidAccessor is created without "wait_for_new_data"
- Throws in read if called on a VoidAccessor without "wait_for_new_data". Wait, this should be write-only in the catalogue. Ah, because we want to throws as early as possible this is already done in the Abstractor.

Reply:
sorry Martin, your comment is confusing... I guess the last part captures what you wanted to say

Reply:
I have to re-read that comment and make a fresh commit.